### ISSUE TITLE
Support for IR Projector

### ISSUE THREAD

**Issue by willliamchan** [OpenWonderLabs/homebridge-switchbot-openapi#54]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

```
[SwitchBot] {"deviceId":"02-202101151706-53910675","deviceName":"Projector","remoteType":"Projector","hubDeviceId":"D949E636A458"}
16/01/2021, 19:19:49 [SwitchBot] Device: Projector with Device Type: Projector, is currently not supported. Submit Feature Requests Here: https://git.io/JL14Z
16/01/2021, 19:19:49 [SwitchBot] {"deviceId":"02-202101161909-84727995","deviceName":"Air Conditioner MB","remoteType":"Air Conditioner","hubDeviceId":"D949E636A458"}
16/01/2021, 19:19:49 [SwitchBot] Device: Air Conditioner MB with Device Type: Air Conditioner, is currently not supported. Submit Feature Requests Here: https://git.io/JL14Zk```

**Comment by donavanbecker** [comment]
@willliamchan Thanks for the feature request. The logs you have provided are the device discovery logs that I am looking for. I will see if I can implement without. If not I will reach back out. This might take a little bit to implement.

**Comment by willliamchan** [comment]
Many thanks!

Regards
William
________________________________

**Comment by donavanbecker** [comment]
@willliamchan,

I have release a potential beta that will work with the projector. it will display as a TV. if you could try it?

**Comment by willliamchan** [comment]
2.0.1.-beta.3 ?

Cheers
William
________________________________

**Comment by donavanbecker** [comment]
> 2.0.1.-beta.3 ? Cheers William

Yes!

**Comment by willliamchan** [comment]
Mate it works!

I got a follow up question for this one. The projector switch on with one set of HEX code (one button), however when I wanna switch it off I will have to send the first HEX code (switch off), and it will pop up a MENU and ask "Yes" / "No", I will then need to send another HEX code (Enter) to confirm the action.

Is there any workaround or should I create a support ticket with SwitchBot support so they might follow up and update the button?

Below is my "Switch Off" action configured in Broadlink RM plug-in. The first HEX is same as the "OFF" button in SwitchBot app, while the second HEX code is "Enter". I put also a 2 seconds pause in between to ensure the second HEX won't send out too fast. You may want to ignore the interval and sendcount as it is some sort of WiFi issue with the Broadlink RM which keep dropping WiFi.

--- START --

"off": [
                    {
                        "data": "260050000001279411141213111412131114111510141138113812131312123811381114121312131213111412131115103812141114111412371138123712381114123712371337120005b20001254b12000d050000000000000000",
                        "interval": 0.5,
                        "pause": 2,
                        "sendCount": 3
                    },
                    {
                        "data": "260050000001249511141115101510141115111311151137123910141214103812371214101510151114111411141114113811141237111411391138113812371213123811151137120005b20001264a12000d050000000000000000",
                        "interval": 0.5,
                        "pause": 2,
                        "sendCount": 3
                    }
                ]
            },

--- END ---

Cheers
William
________________________________

**Comment by donavanbecker** [comment]
> Mate it works! I got a follow up question for this one. The projector switch on with one set of HEX code (one button), however when I wanna switch it off I will have to send the first HEX code (switch off), and it will pop up a MENU and ask "Yes" / "No", I will then need to send another HEX code (Enter) to confirm the action. Is there any workaround or should I create a support ticket with SwitchBot support so they might follow up and update the button?

If I would be able to implement something like this, how would you expect it to show up in the Home app?

**Comment by willliamchan** [comment]
I think we can keep it as a ON/OFF switch with a TV icon.

Regards
William
________________________________

**Comment by donavanbecker** [comment]
As you probably already know, this has been released with v2.1.0
